crates/cubecl-cpp/src/shared/base.rs

```diff
@@ -1056,17 +1138,52 @@ impl<D: Dialect> CppCompiler<D> {
                 instructions.push(Instruction::Neg(self.compile_unary(op, out)))
             }
             gpu::Arithmetic::Normalize(op) => {
-                instructions.push(Instruction::Normalize(self.compile_unary(op, out)))
+                let op = self.compile_unary(op, out);
+                instructions.push(self.select_fast_float(
+                    out.ty,
+                    modes,
+                    FastMath::ReducedPrecision | FastMath::NotNaN | FastMath::NotInf,
```

Comment:
Why is there fast math hardcoded options here?

Reply:
Those are the modes required for the intrinsic. CUDA doesn't have fine-grained settings like Vulkan, so `select_fast_float` checks if the required flags are set and then switches to the intrinsic, otherwise the default implementation. You can take a look at the args for `select_fast_float`.